src/common/translations/localize.ts

```diff
@@ -116,13 +94,29 @@ export const computeLocalize = async <Keys extends string = LocalizeKeys>(
       } catch (err: any) {
         return "Translation error: " + err.message;
       }
-      cache._localizationCache[messageKey] = translatedMessage;
+      localizationCache[translatedValue] = translatedMessage;
     }
 
     let argObject = {};
+    // TS says that we will always get the new format, but we might get the old format
+    const args = _args as any;
     if (args.length === 1 && typeof args[0] === "object") {
       argObject = args[0];
-    } else {
+      if (Object.values(argObject).some((v) => v === undefined)) {
```

Comment:
I don't like that we are now checking all the parameters passed everytime honestly, this is just hurting performance for not much reason I feel.

Reply:
Negligible. This only adds 1 ms to run time for every 13,000 times `localize` is called.